pytestqt/qtbot.py

```diff
@@ -315,10 +317,19 @@ def waitSignals(self, signals=None, timeout=1000, raising=None):
            blocker.wait()
 
         :param list signals:
-            A list of :class:`Signal` objects to wait for. Set to ``None`` to
-            just use timeout.
+            A list of :class:`Signal`s objects to wait for. Set to ``None`` to just use
+            timeout.
         :param int timeout:
             How many milliseconds to wait before resuming control flow.
+        :param str force_order:
+            "none": no order is enforced
+            "strict": the signals have to be emitted in the provided order (and no intermediate signals may be emitted)
+            "simple": like "strict", but signals may be emitted inbetween the provided ones, e.g. suppose
+            signals=[a, b, c] and the tested object emits the signals a, a, b, a, c, the test will pass with "simple"
+            but not with "strict"
+        :param list callbacks:
+            optional list of Callable functions that evaluate the parameters of the signal. Each element of the list
```

Comment:
So, here is a suggestion:
```
:param list check_param_callbacks: list of callables that compare the provided signal parameter(s) to some expected parameter. Each callable has to match the signature of the corresponding signal in ``signals`` (just like a slot function would) and return ``True`` if parameters match, ``False`` otherwise. Instead of a specific callable, ``None`` can be provided, to disable parameter checking for the corresponding signal. If the number of callbacks doesn't match the number of signals ``ValueError`` will be raised.
```